test/cpp/api/functional.cpp

```diff
@@ -88,7 +88,7 @@ TEST_F(FunctionalTest, CosineSimilarity) {
 }
 
 TEST_F(FunctionalTest, SoftMarginLossDefaultOptions) {
-  auto input = torch::tensor({2., 4., 1., 3.}, torch::requires_grad());
+  auto input = torch::tensor({2., 4., 1., 3.}, torch::dtype(torch::kFloat).requires_grad(true));
```

Comment:
I explicitly passed the dtype to all `torch::tensor` calls that use options, so that the resulting tensor's dtype is always what we expect it to be.